### DISCUSSION TITLE
add easier way to change default/fallback language behaviour

### DISCUSSION THREAD

**Discussion by k4lizen** [helix-editor/helix#11827] [General]
### Motivation
I often edit files which dont have a file type nor shebang (like ~/.gdbinit) and for me they most often use `#` for comments. Programming languages which use `//` in general have a file extension so are detected. Maybe `#` should be the default?
### My solution
I want to make it so that `#` is used for comments by default when a specific language isn't detected.
### The problem
This isn't intuitive at all, and reading the docs doesn't help much. It seems the current solution is doing this: https://github.com/helix-editor/helix/issues/9165#issuecomment-2310454339 which seems way to hacky and verbose to overwrite a simple configuration option.
I would expect:
```toml
[[language]]
name = "text"
comment-tokens = ["#"]
```
or something similar to Just Work.

**Comment by Flip1** [comment]
Helix is ​​really great, but I'm failing here too and can only get ahead with workarounds.

Overwriting the default configuration for "text" also seems to be the best solution to me.

Another possibility could be to use "" for file-types, to apply to all files that don't have a suffix.
```
[[language]]
name = "text"
comment-tokens = "#"
file-types = ["","txt"]
```
As a workaround for dot-files the following is possible:
```
[[language]]
name = "bash"
file-types = [{glob=".*"},"sh"]
```
EDIT: Missing quote added

**Comment by k4lizen** [comment]
> As a workaround for dot-files the following is possible:
[[language]]
name = "bash"
file-types = [{glob=".*},"sh"]

Cool! Thanks, I'll be using that. (sidenote you're missing a doublequote in `file-types = [{glob=".*},"sh"]`)

**Comment by kirawi** [comment]
At best I'm paraphrasing here and at worst I'm completely wrong, but from my understanding the plan is to let this use-case be handled by the plugin system instead of putting more work into supporting changing the defaults.

**Comment by k4lizen** [comment]
Oh boy I can't wait to install the always-use-4-space-indentation plugin, I love installing random software instead of having a single knob to turn!

**Comment by kirawi** [comment]
You won't have to install anything. The configuration will simply be more flexible and still declarative. See https://github.com/helix-editor/helix/discussions/10441 and relatedly https://github.com/helix-editor/helix/issues/8853

**Comment by k4lizen** [comment]
Oh okay, I misunderstood, my bad. The discussions you linked seem quite promising!

**Comment by gyreas** [comment]
> At best I'm paraphrasing here and at worst I'm completely wrong, but from my understanding the plan is to let this use-case be handled by the plugin system instead of putting more work into supporting changing the defaults.

Make sense 'cause one would expect that we don't do clever filetype checking, thus fallback to 'text' as type.

**Comment by k4lizen** [comment]
```toml
[[language]]
name = "bash"
file-types = [{glob=".*"},"sh"]
```
Unfortunately when using this files like `~/.local/lib/python3.12/site-packages/decomp2dbg/clients/gdb/gdb_client.py` seem to be caught as well and the proper LSP (pylsp/ruff) isn't detected. I don't see a good way to get around this by looking at https://docs.helix-editor.com/languages.html#file-type-detection-and-the-file-types-key and https://en.wikipedia.org/wiki/Glob_(programming) either

**Comment by pascalkuthe** [comment]
The ability to more easily configure fallback falls under 5he config system refactor https://github.com/helix-editor/helix/issues/8853 (just liek the issue you linked for a workaround).

**Reply by k4lizen** [reply]
> (just like the issue you linked for a workaround).

oh true I completely blanked on that
